src/confetti.js

```diff
@@ -857,6 +857,39 @@
     };
   }
 
+  function shapesFromImage(imageData) {
+    var scalar = imageData.scalar != null ? imageData.scalar : 1;
+    var size = 10 * scalar;
+
+    var img = new Image();
+    img.src = imageData.src;
+
+    return promise(function (resolve) {
+      img.addEventListener('load', function() {
+        var sprites = imageData.sprites != null
+          ? imageData.sprites
+          : [{ x: 0, y: 0, width: img.naturalWidth, height: img.naturalHeight }];
+
+        var baselineWidth = Math.max.apply(null, sprites.map(function(rect) { return rect.width; }));
+
+        resolve(sprites.map(function(sprite) {
+          var width = size * sprite.width / baselineWidth;
+          var height = size * sprite.height / baselineWidth;
+          var scale = 1 / scalar;
+
+          var canvas = new OffscreenCanvas(width, height);
+          canvas.getContext('2d').drawImage(img, sprite.x, sprite.y, sprite.width, sprite.height, 0, 0, width, height);
+
+          return {
+            type: 'bitmap',
+            bitmap: canvas.transferToImageBitmap(),
+            matrix: [scale, 0, 0, scale, -width * scale / 2, -height * scale / 2]
+          };
+        }));
```

Comment:
Okay, so some info about the original sprite issue, since I know I didn't document much in the original tasks.

Once upon a time, there was the crazy idea of rendering emoji. This was pretty incompatible with how rectangles and circles are rendered -- since those are point-based, we can calculate where the points would exist if a 3D transformation was applied, and then do very single (read: cheap) 2D objects with silly points. However, at the time, neither text nor images seemed like they were going to be that simple. They are not point based, so I can't just manipulate their points. At the time, the only method I could find render emoji was to use transforms on the canvas, so rather than applying a 3D effect on the confetti, the 3D effect is applied to the canvas and the confetti was drawn normally. This is incredibly expensive and performance was abysmal, so I did not want to put in in production.

I know that in game rendering (at least back in the day), it was not unheard of to have a sprite for any given asset, and that sprite is many many different 2D images of that resource from all possible angles. This means any time you need to render that asset, you just pick the correct angle out of the sprite -- super cheap. So by saying "support sprites", what I meant was support a way to pre-render the entire animation sequence for a single piece of confetti, and then render each frame as a portion of that sprite.

That is all a long-winded way (sorry) of saying: sprites here complicate the API without delivering any benefit. I would not support them... at least not this version. A user just as well could create multiple shapes from multiple separate images (or honestly, just create the shape in a loop and apply cropping each time if for some reason they must load a sprite image of completely separate objects).